lib/surface/compiler/eex_engine.ex

```diff
@@ -399,16 +400,10 @@ defmodule Surface.Compiler.EExEngine do
     !!value
   end
 
-  defp to_prop_expr([%AST.AttributeExpr{value: expr}], :boolean) do
-    quote generated: true do
-      !!unquote(expr)
-    end
+  defp to_prop_expr([%AST.AttributeExpr{value: value, meta: meta}], type) do
+    Surface.TypeHandler.update_prop_expr(type, value, meta)
   end
 
-  defp to_prop_expr([%AST.AttributeExpr{value: {_, value}}], :list), do: value
-
-  defp to_prop_expr([%{value: value}], :string), do: value
-
   defp to_prop_expr(values, :string) do
```

Comment:
This variant should also be removed once we combine all values in a list into a single